src/rest_framework_dso/fields.py

```diff
@@ -71,12 +71,9 @@ def __repr__(self):
         else:
             return "<FieldsToDisplay: deny all>"
 
-    def __bool__(self):
-        """Whether the returned fields need to be reduced.
-        Note that this returns false when there are child nodes that need to be reduced.
-        These are found with :meth:`allow_nested`, :meth:`as_nested` and :attr:`children`.
-        """
-        return not self._allow_all
+    def allow_all(self):
+        """Returns false when there are child nodes that need to be reduced, else true."""
+        return self._allow_all
```

Comment:
This breaks things, e.g. `elif nested_fields_to_display:`. And leaving the `__bool__()` in there won't harm.